### PR TITLE
fixes an error where email.raw was too large

### PR DESCRIPTION
**Pull request by Regan831**

### CODE REVIEW

File: lib/mailserver.js [hunk 1]
```diff
@@ -32,10 +32,10 @@ var elasticsearch = require('elasticsearch');
 
 var mailServer = module.exports = {};
 
-mailServer.elasticsearchClient = null;
+mailServer.elasticsearchClient = 'http:localserver:9200';
```
Comment (sarahleeashraf) [inline lib/mailserver.js]: Can we handle these in the else case in the listen? That way all the logic is in the same place. 

https://github.com/Regan831/MailDev/blob/dcd7a3d64e4bd954aa68c9f0e28b1fcb2d936205/lib/mailserver.js#L185

File: lib/mailserver.js [hunk 2]
```diff
@@ -119,47 +119,50 @@ function saveEmail(id, envelope, mailObject) {
...
         from: connection.from,
         to: connection.to,
         host: connection.host,
         remoteAddress: connection.remoteAddress
     }));
-}
+};
 
-function writeChunk(connection, chunk){
+mailServer.createNewMailParser = function() {
```
Comment (sarahleeashraf) [inline lib/mailserver.js]: let's add comments for anything new we add.

File: test/mailServerTest.js [hunk 3]
```diff
@@ -553,4 +553,167 @@ describe('setup for all tests', function () {
...
+                remoteAddress: 'address'};
+
+            monkeyPatcher.patch(mailserver, 'makeId', function () {
+                return 'ABCD1234';
+            });
+
+            monkeyPatcher.patch(mailserver, 'createNewMailParser', function (){
+                return {
+                    mailparser: 'New MailParser',
+                    on: function () {
```
Comment (sarahleeashraf) [inline test/mailServerTest.js]: We should test what is being passed on 'on' as well

File: test/mailServerTest.js [hunk 4]
```diff
@@ -553,4 +553,167 @@ describe('setup for all tests', function () {
...
+               to: ['13592985+99323@signpost.com'],
+               date: 'Fri Sep 16 2016 11:04:44 GMT-0500 (CDT)',
+               remoteAddress: 'address',
+               saveStream: {
+                   write: function (chunk) {
+                       return chunk;
+                   }
+               }
+           };
+
```
Comment (sarahleeashraf) [inline test/mailServerTest.js]: we should also assert the rawString gets the chunk appended.

File: test/mailServerTest.js [hunk 5]
```diff
@@ -553,4 +553,167 @@ describe('setup for all tests', function () {
...
+                   type: 'email',
+                   id: 'testId',
+                   body: {
+                       raw: 'Raw String'
+                   }
+               });
+
+                return Q.resolve(true);
+            });
+
```
Comment (sarahleeashraf) [inline test/mailServerTest.js]: We should also assert that rawString is reset here.

File: test/mailServerTest.js (hunk 3 above)
Comment (Regan831) [inline test/mailServerTest.js]: How can I test this?

Comment (sarahleeashraf) [inline test/mailServerTest.js]: you can put an assertion in here so like
```on: function(eventName, callback) {
  assert.equal(eventName, 'something');
  assert.equal(callback, 'some other thing');
}```

File: lib/mailserver.js [hunk 6]
```diff
@@ -32,10 +32,10 @@ var elasticsearch = require('elasticsearch');
 
 var mailServer = module.exports = {};
 
-mailServer.elasticsearchClient = null;
+
 
 // should be overrode in listen()
-mailServer.elasticsearchConfig = null;
+mailServer.elasticsearchConfig = { host: 'http:localserver:9200'};
```
Comment (sarahleeashraf) [inline lib/mailserver.js]: can we move this to the else as well?

File: test/mailServerTest.js [hunk 7]
```diff
@@ -553,4 +553,189 @@ describe('setup for all tests', function () {
...
+                return true;
+            });
+
+            monkeyPatcher.patch(mailserver, 'createNewMailParser', function (){
+                return {
+                    mailparser: 'New MailParser',
+                    on: function (eventName, callback) {
+                        assert.equal(eventName, 'end');
+                        callback();
```
Comment (sarahleeashraf) [inline test/mailServerTest.js]: This test is kind of hard to follow. I think some comments would be helpful.

File: test/mailServerTest.js (hunk 5 above)
Comment (sarahleeashraf) [inline test/mailServerTest.js]: looks like we also need to assert that done gets called? Can we use that to bring the rawString assertion back into the main part of the test?

File: test/mailServerTest.js [hunk 8]
```diff
@@ -553,4 +553,189 @@ describe('setup for all tests', function () {
...
+
+    describe('mailserver.createElasticsearchClient', function () {
+        it('Should create a new elasticsearch client when given a host', function (done) {
+            mailserver.createElasticsearchClient('http://localhost:9200', 'trace');
+            done();
+        })
+    });
+
+    describe('mailserver.saveEmail', function () {
+        it('Should save an email sent to maildev', function (done) {
```
Comment (sarahleeashraf) [inline test/mailServerTest.js]: we should also assert that whatever object is supposed to be emitted is actually emitted.
